Explorer/Assets/DCL/Profiles/RealmProfileRepository.cs

```diff
@@ -0,0 +1,203 @@
+using CommunicationData.URLHelpers;
+using Cysharp.Threading.Tasks;
+using DCL.WebRequests;
+using ECS;
+using Ipfs;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace DCL.Profiles
+{
+    public class RealmProfileRepository : IProfileRepository
+    {
+        private static readonly JsonSerializerSettings SERIALIZER_SETTINGS = new () { Converters = new JsonConverter[] { new ProfileJsonRootDtoConverter() } };
+
+        private readonly IWebRequestController webRequestController;
+        private readonly IRealmData realm;
+        private readonly IProfileCache profileCache;
+        private readonly URLBuilder urlBuilder = new ();
+
+        public RealmProfileRepository(IWebRequestController webRequestController,
+            IRealmData realm,
+            IProfileCache profileCache)
+        {
+            this.webRequestController = webRequestController;
+            this.realm = realm;
+            this.profileCache = profileCache;
+        }
+
+        public async UniTask<Profile?> GetAsync(string id, int version, CancellationToken ct)
+        {
+            if (string.IsNullOrEmpty(id)) return null;
+
+            Profile? profileInCache = profileCache.Get(id);
+
+            if (profileInCache?.Version > version)
+                return profileInCache;
+
+            IIpfsRealm ipfs = realm.Ipfs;
+
+            urlBuilder.Clear();
+
+            urlBuilder.AppendDomain(ipfs.LambdasBaseUrl)
+                      .AppendPath(URLPath.FromString($"profiles/{id}"))
+                      .AppendParameter(new URLParameter("version", version.ToString()));
+
+            URLAddress url = urlBuilder.Build();
+
+            try
+            {
+                GenericGetRequest response = await webRequestController.GetAsync(new CommonArguments(url), ct);
+
+                using GetProfileJsonRootDto root = await response.CreateFromNewtonsoftJsonAsync<GenericGetRequest, GetProfileJsonRootDto>(
+                    createCustomExceptionOnFailure: (exception, text) => new ProfileParseException(id, version, text, exception),
+                    serializerSettings: SERIALIZER_SETTINGS);
+
+                if (root.avatars == null) return null;
+                if (root.avatars.Count == 0) return null;
+
+                Profile profile = profileInCache ?? new Profile();
+                root.avatars[0].CopyTo(profile);
+                profileCache.Set(id, profile);
+
+                return profile;
+            }
+            catch (UnityWebRequestException e)
+            {
+                if (e.ResponseCode == 404)
+                    return null;
```

Comment:
I understand the intention of returning `null` if `Profile` is not found but we should be very careful with this as it's not clear if can ever handle nulls. It will be clear when more use cases of this method emerge.